### PR TITLE
REST API to ADD and READ a product

### PR DESCRIPTION
**Pull request by zqf0722**
- CREATE a product and add it into a shopcart
- READ a product in a shopcart
- Fix an error in the shopcart serialization. Now it can be correctly serialize to JSON format

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -18,17 +18,142 @@
...
     )
 
 
+######################################################################
+# REST API
+######################################################################
+
+# -----------------------------------------------------------
+# Create shopcarts
+# -----------------------------------------------------------
```
Comment (Yuhao-Fan) [inline service/routes.py:32]: Can you change this 2 comments into the same form

File: service/routes.py [hunk 2]
```diff
@@ -18,17 +18,142 @@
...
+    # Return the new shopcart
+    app.logger.info("Returning shopcart: %s", user_id)
+    return jsonify(shopcarts[0].serialize()), status.HTTP_200_OK
+
+######################################################################
+# LIST ALL SHOPCARTS
+######################################################################
+@app.route("/shopcarts", methods=["GET"])
+def list_shopcarts():
+    """Returns all of the Shopcarts"""
```
Comment (Yuhao-Fan) [inline service/routes.py:91]: Can you add args and returns like other functions?

**Review by Yuhao-Fan — APPROVED** [review]
Please Keep the comment form consistent.

File: service/routes.py (hunk 2 above)
Comment (zqf0722) [reply service/routes.py:91]: Got it. I'll raise a new PR for this

**Pull request merged**